blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSuite.scala

```diff
@@ -157,15 +157,6 @@ class Http1ClientStageSuite extends Http4sSuite {
     } yield ()).intercept[Http1Connection.InProgressException.type]
   }
 
-  fooConnection.test("Reset correctly") { tail =>
```

Comment:
So to re-echo my earlier comment, there is no guarantee that the connection is ready to be reused after the read finishes since the write flag can still be set to true. There's likely a fairness penalty here, but it solves the race condition.